src/overseer/store/jdbc.clj

```diff
@@ -166,9 +172,15 @@
                         (map job->jdbc-map)
                         (map #(assoc % :created_at now :updated_at now)))
           dep-rows (job-dep-jdbc-maps graph)]
-      (j/with-db-transaction [t-con db-spec]
-        (j/insert-multi! t-con :overseer_jobs job-rows)
-        (j/insert-multi! t-con :overseer_dependencies dep-rows))))
+      (try
+        (j/with-db-transaction [t-con db-spec]
+          (j/insert-multi! t-con :overseer_jobs job-rows)
+          (j/insert-multi! t-con :overseer_dependencies dep-rows))
+        graph
+        (catch Exception ex
```

Comment:
So this is an interesting thought, except I think this only really makes sense if you have an enum / mapping for _every possible exception_ , otherwise we're in this odd place of `PrimaryKeyException | AnythingElse`. We could also possibly try

```java
public class OverseerPrimaryKeyException extends java.sql.SQLException {}

public SQLException normalize(SQLException e) {
  if (is_mysql_or_h2_pk_violation(ex)) {
    return new OverseerPrimaryKeyException(ex);
  } else {
    return ex;
  }
}
```

but TBH I think both of these ventures are more work than they're worth.

What we can easily do is at least scope down the MySQL case and look for a `com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException` directly instead of examining a string message. H2 appears to throw a generic SQLException so I don't know how much luck we'll have there.